src/core/border_router/routing_manager.cpp

```diff
@@ -3676,9 +3690,12 @@ void RoutingManager::PdPrefixManager::ProcessPlatformGeneratedRa(const uint8_t *
     Error                     error = kErrorNone;
     RouterAdvert::Icmp6Packet packet;
 
-    VerifyOrExit(IsRunning(), LogWarn("Ignore platform generated RA since PD is disabled or not running."));
     packet.Init(aRouterAdvert, aLength);
-    error = Process(RouterAdvert::RxMessage(packet));
+    RouterAdvert::RxMessage aMessage = RouterAdvert::RxMessage(packet);
+
+    VerifyOrExit(mEnabled, LogWarn("Ignore platform generated RA since PD is disabled."));
```

Comment:
Do we have any reason for moving this check? I would prefer checking the state at the entrance of the function.

Also, I see you are using `mEnabled`, do we have any concern not using `IsRunning()`?

When there are multiple BRs in the network, only one of the BRs should request prefix, using `IsRunning()` will generate a warning message indicating "the device is still requesting PD prefix while we should not do so"

Reply:
I had to move the check because I was getting compiler errors. 

- I can't do `Process(&RouterAdvert::RxMessage(packet), nullptr);` - compiler error
- I cannot define `RouterAdvert::RxMessage aMessage` next to ` RouterAdvert::Icmp6Packet packet` because `RouterAdvert::RxMessage` has no default constructor.
- I cannot define `RouterAdvert::RxMessage aMessage = RouterAdvert::RxMessage(packet);` after `VerifyOrExit` - compiler error  `error: jump to label 'exit crosses initialization of 'ot::Ip6::Nd::RouterAdvert::RxMessage aMessage`

I updated the code to check the state at the entrance of the function using `if` instead of `VerifyOrExit` as I couldn't find a better option.

I discussed the second point with Abtin in [this comment ](https://github.com/openthread/openthread/pull/10000#discussion_r1556443620). I don't think there should be a problem with this change as the received prefix is set in `mPrefix` but no action will be taken until a policy evaluation. The policy evaluation will trigger only when `IsRunning()` is true.